overlord/ifacestate/handlers.go

```diff
@@ -81,20 +81,51 @@ func getExtraLayouts(st *state.State, snapInfo *snap.Info) ([]snap.Layout, error
 	return extraLayouts, nil
 }
 
-func buildConfinementOptions(st *state.State, snapInfo *snap.Info, flags snapstate.Flags) (interfaces.ConfinementOptions, error) {
+func buildConfinementOptions(st *state.State, snapInfo *snap.Info, flags snapstate.Flags, usePromptPrefix func(task *state.Task) bool, task *state.Task) (interfaces.ConfinementOptions, error) {
 	extraLayouts, err := getExtraLayouts(st, snapInfo)
 	if err != nil {
 		return interfaces.ConfinementOptions{}, fmt.Errorf("cannot get extra mount layouts of snap %q: %s", snapInfo.InstanceName(), err)
 	}
 
+	if usePromptPrefix == nil {
+		// Get a usePromptPrefix closure which will use the "use-prompt-prefix"
+		// value from the given task, if it exists, or check the current state
+		// as to whether prompting is supported and enabled.
+		usePromptPrefix = buildDefaultUsePromptPrefixFunc(st)
+	}
+
 	return interfaces.ConfinementOptions{
-		DevMode:      flags.DevMode,
-		JailMode:     flags.JailMode,
-		Classic:      flags.Classic,
-		ExtraLayouts: extraLayouts,
+		DevMode:           flags.DevMode,
+		JailMode:          flags.JailMode,
+		Classic:           flags.Classic,
+		ExtraLayouts:      extraLayouts,
+		AppArmorPrompting: usePromptPrefix(task),
 	}, nil
 }
 
+func buildDefaultUsePromptPrefixFunc(st *state.State) func(t *state.Task) bool {
```

Comment:
Why is this required?

Could we just store the value and respect it when set, defaulting to off?